worker/deployer/nested.go

```diff
@@ -50,10 +50,11 @@ type Logger interface {
 var _ Context = (*nestedContext)(nil)
 
 type nestedContext struct {
-	logger            Logger
-	agentConfig       agent.Config
-	baseUnitConfig    UnitAgentConfig
-	updateConfigValue func(string, string) error
+	logger Logger
+	agent  agent.Agent
+	// agentConfig is a snapshot of the current configuration.
+	agentConfig    agent.Config
```

Comment:
Would it be worth naming this to `agentConfigSnapshot` based on the comment?

Reply:
I don't think so. The comment was really a reminder for me as it was the cause of a bug. Now it should be obvious in the usage for the package as it is created with the `CurrentConfig` method which clearly mentions that it is a copy of the config.